File: Data Structures & Algorithms/string-encode-and-decode/submission-1.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string ans = "";

        for (string s : strs) {
            ans += to_string(s.length()) + "#" + s;
        }

        return ans;
    }

    vector<string> decode(string s) {
        vector<string> ans;
        int i = 0;

        while (i < s.length()) {
            int j = i;

            // find '#'
            while (s[j] != '#') j++;

            int len = stoi(s.substr(i, j - i));

            // extract string of length len
            string word = s.substr(j + 1, len);
            ans.push_back(word);

            // move pointer
            i = j + 1 + len;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-1.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n + 1];

        int ind = 0;

        for (int i = 0; i <= n; i++) {
            int cnt = 0;
            int x = i;

            while (x > 0) {
                if ((x & 1) == 1) cnt++;
                x = x >> 1;
            }

            ans[ind] = cnt;
            ind++;
        }
        return ans;
    }
}
